### DISCUSSION TITLE
Where are the default tests select which will be run after "normal" start?

### DISCUSSION THREAD

**Discussion by oed-streicherd** [memtest86plus/memtest86plus#57] [Q&A] [answered]
Hi,
is there a way to select the tests which will run automatically after booting?
(In the code at compile time or with boot params?)

Like I want to start automatic: ALL CPU PAR + Tests 6,7,9

Best regards
Daniel

**Comment by debrouxl** [comment, marked as answer]
All CPU + PAR is currently triggered by the "smp" command-line option, though that might change if/when E-Cores are disabled by default (issue #9 ).
However, AFAICT, non-interactive test list selection is currently done only at compile time, through `tests/tests.c::test_list`. Command-line test list selection would be implemented in `app/config.c::parse_option()` and presumably a static subroutine thereof for clarity, like `parse_serial_params()`.

**Comment by oed-streicherd** [comment]
Thanks for the hint, got it working for me:
![image](https://user-images.githubusercontent.com/102787488/165945753-cf221bb4-d2f7-4dfd-8a1d-36681c8cac62.png)
tests.c Code false/code at selection:
![image](https://user-images.githubusercontent.com/102787488/165945913-11bac369-99b3-46fc-a8eb-f957da881967.png)
